smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -22,53 +22,22 @@ contract MixinPurchase is
   using SafeMath for uint;
 
   /**
-  * @dev Purchase function, public version, with no referrer.
-  * @param _recipient address of the recipient of the purchased key
-  */
-  function purchaseFor(
-    address _recipient
-  )
-    external
-    payable
-    onlyIfAlive
-  {
-    return _purchaseFor(_recipient, address(0));
-  }
-
-  /**
-  * @dev Purchase function, public version, with referrer.
+  * @dev Purchase function
   * @param _recipient address of the recipient of the purchased key
   * @param _referrer address of the user making the referral
-  */
-  function purchaseForFrom(
-    address _recipient,
-    address _referrer
-  )
-    external
-    payable
-    onlyIfAlive
-    hasValidKey(_referrer)
-  {
-    return _purchaseFor(_recipient, _referrer);
-  }
-
-  /**
-  * @dev Purchase function: this lets a user purchase a key from the lock for another user
-  * @param _recipient address of the recipient of the purchased key
   * This will fail if
   *  - the keyReleaseMechanism is private
   *  - the keyReleaseMechanism is Approved and the recipient has not been previously approved
   *  - the amount value is smaller than the price
   *  - the recipient already owns a key
-  * TODO: next version of solidity will allow for message to be added to require.
   */
-  function _purchaseFor(
+  function purchase(
     address _recipient,
     address _referrer
```

Comment:
So fo r a purchase with no referrer, a 0 has to be passed in as the second param here right? I'm on board with pushing that responsibility to the front end in order to simplify things in the contract.

Reply:
That's correct